### PR TITLE
Feature/6/update firestore rules

### PR DESCRIPTION
**Pull request by GabeDiniz**
**⭐ What was done:**
1. Update Firestore rules to allow WRITE only if USER is authenticated.
2. ADMIN account can bypass and WRITE to all collections
3. Regular users can only WRITE to resources they own → UPDATING their profile details etc..
4. To read from Firestore, all users need to be authenticated. When reading from private or sensitive data, such as user data, only the owner of the resources is allowed to read it.
5. Added Role Claim functions

**📝Note:**
The Functions were deployed already and can be found in the HawkHacks dashboard: https://console.firebase.google.com/u/0/project/hawkhacks-dashboard/functions

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -28,11 +32,12 @@ module.exports = {
     plugins: ["@typescript-eslint", "react"],
     rules: {
         indent: ["error", 4],
-        "linebreak-style": ["error", "unix"],
+        "linebreak-style": ["error", "linux"],
```
Comment (juancwu) [inline .eslintrc.cjs:35]: Is there a reason why it needs to be `linux`? This is causing the eslint action to fail.

File: functions/package-lock.json
Comment (juancwu) [inline functions/package-lock.json:1]: This file shouldn't be here since we are using `pnpm`

**Review by juancwu — CHANGES_REQUESTED** [review]

File: .eslintrc.cjs (hunk 1 above)
Comment (GabeDiniz) [reply .eslintrc.cjs:35]: No, sorry about that. I was getting errors on my end so I changed it to Windows (which fixed it). When I changed it back, I mistakenly changed it to linux when it should have been unix.

File: functions/package-lock.json
Comment (GabeDiniz) [reply functions/package-lock.json:1]: Thank you, fixed that!

**Review by juancwu — APPROVED** [review]

File: functions/tsconfig.json [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
-        "outDir": "lib",
+        "outDir": "./lib",
         "sourceMap": true,
         "strict": true,
         "target": "es2017"
     },
     "compileOnSave": true,
-    "include": ["src"]
+    "include": ["src", "src/**/*.ts", "src/*.ts"],
+    "exclude": ["node_modules"]
```
Comment (aidantrabs) [inline functions/tsconfig.json:13]: are you sure the `node_modules` are being excluded? 

- if yes, this is fine
- if no, include:
`"compilerOptions": {
    "skipLibCheck": true,
    ...
 }`

Comment (GabeDiniz) [reply functions/tsconfig.json:14]: Yea, I should have added the skipLibCheck. I added it and merged it. Thanks Aidan!

**Pull request merged**